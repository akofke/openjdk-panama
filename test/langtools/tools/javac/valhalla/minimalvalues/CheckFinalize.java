/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support finalize
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckFinalize.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror -Xlint:values  CheckFinalize.java
 */
@jdk.incubator.mvt.ValueCapableClass
final class CheckFinalize {
    @Override
    protected void finalize() {} // <-- error

    @jdk.incubator.mvt.ValueCapableClass
    final class CheckFinalizeInner {}

    void foo(CheckFinalizeInner cfi, CheckFinalize cf) {
        cfi.finalize();          // Error
        cf.finalize();           // OK.
    }
}
