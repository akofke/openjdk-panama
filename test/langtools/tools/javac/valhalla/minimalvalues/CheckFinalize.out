- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckFinalize.java:16:12: compiler.err.report.access: finalize(), protected, java.lang.Object
CheckFinalize.java:10:20: compiler.warn.value.does.not.support: overriding java.lang.Object's method: finalize
CheckFinalize.java:16:21: compiler.warn.value.does.not.support: calling java.lang.Object's method: finalize
- compiler.note.deprecated.filename: CheckFinalize.java
- compiler.note.deprecated.recompile
2 errors
3 warnings
